arm-software/embedded/llvmlibc-support/crt0/bootcode.cpp

```diff
@@ -77,13 +77,17 @@ extern "C" void __startup() {
   exit(main(0, 0));
 }
 
+extern "C"
 // The entry point sets sp and branches to the main startup function.
 #ifdef __ARM_ARCH_ISA_ARM
 // If the target supports the A32 instruction set then the entry point has to
 // use it.
-__attribute__((target("arm")))
+[[gnu::target("arm")]]
+#elif defined(__ARM_ARCH_ISA_A64)
+// AArch64 requires this to be placed before the vector table at `.vectors`
+[[gnu::section(".text.init.enter")]]
```

Comment:
I'm surprised that this needs to be before the vector table in the general case. There's nothing in the architecture that I know of that requires this.

I can understand that when running the sample from a bin file (ELF file can supply model with an entry point), then this bit of code needs to be placed at the address where the model will execute after boot. With the current linker script that may need to be the first address in boot_flash.

If I'm right it will be worth being more precise in the comment.